src/vs/workbench/services/textMate/browser/textMateTokenizationFeatureImpl.ts

```diff
@@ -365,6 +375,44 @@ export class TextMateTokenizationFeature extends Disposable implements ITextMate
 			return response;
 		}
 	}
+
+	public reportTokenizationTime(timeMs: number, languageId: string, sourceExtensionId: string | undefined, lineLength: number, fromWorker: boolean): void {
+		// 50 events per hour (one event has a low probability)
+		if (TextMateTokenizationFeature.reportTokenizationTimeCounter > 50) {
+			// Don't flood telemetry with too many events
+			return;
+		}
+		if (TextMateTokenizationFeature.reportTokenizationTimeCounter === 0) {
+			setTimeout(() => {
+				TextMateTokenizationFeature.reportTokenizationTimeCounter = 0;
+			}, 1000 * 60 * 60);
+		}
+		TextMateTokenizationFeature.reportTokenizationTimeCounter++;
+
+		this._telemetryService.publicLog2<{
+			timeMs: number;
+			languageId: string;
+			lineLength: number;
+			fromWorker: boolean;
+			sourceExtensionId: string | undefined;
+		}, {
+			owner: 'hediet';
+
+			timeMs: { classification: 'SystemMetaData'; purpose: 'FeatureInsight'; isMeasurement: true; comment: 'To understand how long it took to tokenize a random line' };
```

Comment:
This seems really hard to read with larger events like this we normally just make local types above the public log so it's a bit cleaner.